source/Octopus.Tentacle/Commands/RunAgentCommand.cs

```diff
@@ -146,7 +146,7 @@ protected override void Stop()
                 halibut.Value.Stop();
             }
 
-            workspaceCleanerTask.Stop();
+            workspaceCleanerTask.Value.Stop();
```

Comment:
If `RunAgentCommand.Stop` is called before the dependencies are setup, we could crash in the same way we did in the constructor.

Do we think this is possible? Or are you happy that this is a situation that cannot happen?

Reply:
I think the safest way to go for now is just to have a check, like `halibutHasStarted`. I do think it would be safe to assume that if the "has started" flag is true, then there should be a `Value`. We'll take another look with sc-61799. Let me know if you have any concerns though 👂